common/djangoapps/student/views/login.py

```diff
@@ -386,6 +386,28 @@ def send_reactivation_email_for_user(user):
     return JsonResponse({"success": True})
 
 
+@ensure_csrf_cookie
+def verify_user_password(request):
+    """
+    If the user is logged in and we want to verify that they have submitted the correct password
+    for a major account change (for example, deleting this user account).
+
+    Args:
+        request (HttpRequest): A request object where the password should be included in the POST fields.
+    """
+    # TODO: Use the rate limiting code from above.
```

Comment:
What is the plan for locking out the user after exceeding a number of failed attempts?  Both this endpoint and the `login` endpoint need to use the same lockout mechanism and counter.  We have to make sure a malicious attacker doesn't use this endpoint to guess a user's password.

Reply:
Sorry, I did some investigation and didn't remove this comment. By default, the `authenticate` call goes to the same rate limiting backend that the `login` endpoint does, so we get the rate limiting for 'free', so to speak.